RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc

```diff
@@ -568,6 +614,39 @@ void ReducedEGProducer::produce(edm::Event& theEvent, const edm::EventSetup& the
 
   for (reco::GsfElectron &gsfElectron : *gsfElectrons) {
     relinkGsfElectronCore(gsfElectron, gsfElectronCoreMap, outgsfElectronCoreHandle);
+
+    // -----
+    // Also in this loop let's reling ambiguous tracks
+    // -----
+    std::vector<reco::GsfTrackRef> ambigTracksInThisElectron;
+    // Here we loop over the ambiguous tracks and save them in a vector
+    for (reco::GsfTrackRefVector::const_iterator igsf = gsfElectron.ambiguousGsfTracksBegin(); igsf != gsfElectron.ambiguousGsfTracksEnd(); ++igsf) {
+      const reco::GsfTrackRef &ambGsfTrack = *igsf;
+      ambigTracksInThisElectron.push_back(*igsf);
+    }
+    
+    // Now we need to clear them (they are the refs to original collection):
+    gsfElectron.clearAmbiguousGsfTracks();
+    
+    // And here we add them back, now from a new reduced collection:
+    for (std::vector<reco::GsfTrackRef>::const_iterator it = ambigTracksInThisElectron.begin(); it != ambigTracksInThisElectron.end(); ++it) {
```

Comment:
Here you are looping over a std::vector<> , and therefore range based loops must be used:
(e.g. http://en.cppreference.com/w/cpp/language/range-for)